default-recommendations/mutability-predicates.rkt

```diff
@@ -0,0 +1,60 @@
+#lang racket/base
+
+
+(require racket/contract/base)
+
+
+(provide
+ (contract-out
+  [mutability-predicates refactoring-suite?]))
+
+
+(require rebellion/private/static-name
+         resyntax/base
+         syntax/parse)
+
+
+;@----------------------------------------------------------------------------------------------------
+
+
+(define-refactoring-rule hash-and-immutable-to-immutable-hash
+  #:description "This contract is equivalent to the `immutable-hash?` predicate."
+  #:literals (and/c hash? immutable?)
+  (~or (and/c hash? immutable?) (and/c immutable? hash?))
+  #:when (identifier-binding (datum->syntax this-syntax 'immutable-hash?))
+  #:with replacement-id (datum->syntax this-syntax 'immutable-hash?)
+  replacement-id)
```

Comment:
You don't need to do this `datum->syntax` stuff. Once you've imported `racket/mutability`, you can just write `immutable-hash?` in the output template without needing the `#:when` or `#:with` pattern directives at all. Same goes for the rules below.